apollo-router/src/plugins/authentication.rs

```diff
@@ -275,33 +273,43 @@ async fn search_jwks(
                 // Note: Matching algorithm parameters may seem unusual, but the appropriate
                 // algorithm details are not structured for easy consumption in jsonwebtoken and
                 // this is the simplest way to determine algorithm family.
-                None => match criteria.alg {
-                    Algorithm::HS256 | Algorithm::HS384 | Algorithm::HS512 => {
-                        if let AlgorithmParameters::OctetKey(_) = key.algorithm {
+                None => match (criteria.alg, &key.algorithm) {
+                    (
+                        Algorithm::HS256 | Algorithm::HS384 | Algorithm::HS512,
+                        AlgorithmParameters::OctetKey(_),
+                    ) => {
+                        key.common.algorithm = Some(criteria.alg);
+                    }
+                    (
+                        Algorithm::RS256
+                        | Algorithm::RS384
+                        | Algorithm::RS512
+                        | Algorithm::PS256
+                        | Algorithm::PS384
+                        | Algorithm::PS512,
+                        AlgorithmParameters::RSA(_),
+                    ) => {
+                        key.common.algorithm = Some(criteria.alg);
+                    }
+                    (Algorithm::ES256, AlgorithmParameters::EllipticCurve(params)) => {
+                        if params.curve == EllipticCurve::P256 {
                             key.common.algorithm = Some(criteria.alg);
-                        } else {
-                            continue;
                         }
                     }
-                    Algorithm::RS256
-                    | Algorithm::RS384
-                    | Algorithm::RS512
-                    | Algorithm::PS256
-                    | Algorithm::PS384
-                    | Algorithm::PS512 => {
-                        if let AlgorithmParameters::RSA(_) = key.algorithm {
+                    (Algorithm::ES384, AlgorithmParameters::EllipticCurve(params)) => {
+                        if params.curve == EllipticCurve::P384 {
                             key.common.algorithm = Some(criteria.alg);
-                        } else {
-                            continue;
                         }
                     }
-                    Algorithm::ES256 | Algorithm::ES384 | Algorithm::EdDSA => {
-                        if let AlgorithmParameters::EllipticCurve(_) = key.algorithm {
+                    (Algorithm::EdDSA, AlgorithmParameters::EllipticCurve(params)) => {
+                        if params.curve == EllipticCurve::Ed25519 {
                             key.common.algorithm = Some(criteria.alg);
-                        } else {
-                            continue;
                         }
                     }
+                    _ => {
+                        tracing::info!(?criteria.alg, ?key.algorithm, "Ignoring unsupported algorithm combination");
```

Comment:
that should not be an info level log. Like, if we get HS256, and have one HMAC key and 2 RSA keys, then that log will be printed twice

Reply:
Should I just ignore that then?